### PR TITLE
Feature profile viewing

### PR DESCRIPTION
**Pull request by jpablo2002**

### CODE REVIEW

**Review by modaserMoj — DISMISSED** [review]
Code looks good it is following all the required clean principles

**Review by modaserMoj — APPROVED** [review]
New changes are appropriate

File: src/main/java/ca/utoronto/fitbook/adapter/web/model/UserProfileModel.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Value
+public class UserProfileModel {
+    String id;
+    String name;
+    int followingSize;
+    int followerSize;
+    String joinDate;
+    List<Post> postList;
+    List<Post> likedPostList;
+    Map<String, Exercise> userExercises;
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/adapter/web/model/UserProfileModel.java:19]: Add some non-null annotations

File: src/main/java/ca/utoronto/fitbook/application/port/out/response/UserProfileResponse.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Value
+public class UserProfileResponse {
+    String id;
+    String name;
+    int followingSize;
+    int followerSize;
+    String joinDate;
+    List<Post> postList;
+    List<Post> likedPostList;
+    Map<String, Exercise> userExercises;
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/application/port/out/response/UserProfileResponse.java:19]: ditto

File: src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        for(Post post : userPosts){
+            List<Exercise> postExercises = loadExerciseListPort.loadExerciseList(post.getExerciseIdList());
+            for(Exercise exercise: postExercises){
+                userExercises.put(exercise.getId(), exercise);
+            }
+        }
+
+        DateFormat dateFormatter = new SimpleDateFormat("MMMM dd, yyyy");
+        String cleanDate = dateFormatter.format(user.getJoinDate());
+
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java:50]: Extra space

File: src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+        return new UserProfileResponse(user.getId(), user.getName(), user.getFollowingIdList().size(), user.getFollowersIdList().size(), cleanDate, userPosts, likedPosts, userExercises);
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java:52]: Break this up with newlines

File: src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+     * @return a profile response with user information
+     */
+    @Override
+    public UserProfileResponse createProfile(UserProfileCommand command) {
+        User user = userProfilePort.loadUser(command.getUserId());
+        List<Post> userPosts = loadPostListPort.loadPostList(user.getPostIdList());
+        List<Post> likedPosts = loadPostListPort.loadPostList(user.getLikedPostIdList());
+        Map<String, Exercise> userExercises = new HashMap<>();
+
+        for(Post post : userPosts){
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java:41]: Add spaces between `for` and `(` -- and -- `)` and `{`

File: src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            List<Exercise> postExercises = loadExerciseListPort.loadExerciseList(post.getExerciseIdList());
+            for(Exercise exercise: postExercises){
```
Comment (youssefsoli) [inline src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java:43]: ditto

File: src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java (hunk 3 above)
Comment (jpablo2002) [reply src/main/java/ca/utoronto/fitbook/application/service/UserProfileService.java:50]: I think I addressed this but if I misunderstood just let me know!

**Review by youssefsoli — APPROVED** [review]
r+

**Review by feiwun — APPROVED** [review]
code follows clean architecture well

**Review by JacketPants12 — APPROVED** [review]
lgtm

**Review by modaserMoj — APPROVED** [review]

**Comment by modaserMoj** [comment]
All seems good

**Pull request merged**